### ISSUE TITLE
Error when using Versioned dataobject without Live stage

### ISSUE THREAD

**Issue by markguinn** [unclecheese/silverstripe-gridfield-betterbuttons#36]
If a dataobject uses the Versioned extension without using Stage/Live, the module fails. The shop module uses this configuration for ProductVariations so they have a version history. In this case the extension is initialized like so:

```
    static $versioning = array(
        'Stage'
    );

    static $extensions = array(
        "Versioned('Stage')"
    );
```

The following error is then given when you try to edit one of these objects from modeladmin with the betterbuttons module installed:

```
[User Error] Uncaught Exception: Object->__call(): the method 'isnew' does not exist on 'ProductVariation'
```

If the isNew method is added to the model, there's still an error caused by the following query (GridFieldBetterButtonsItemRequest.php, 505):

```
        return (bool) DB::query("SELECT \"ID\" FROM \"{$table}_Live\" WHERE \"ID\" = {$this->owner->record->ID}")->value();
```

Basically, the module assumes that the object is a SiteTree if it has a Versioned extension. I tried to find a workaround and couldn't find a satisfactory one. Versioned apparently doesn't have any way for an outside object to find out whether the Live stage is enabled or not, which seems to be a weakness in it's api.

The only solution I can think of is to check for SiteTree instead of Versioned and/or to submit a pull to silverstripe-framework with a more complete api for Versioned.
